### PR TITLE
Add test_assignment fixture and simple test

### PR DESCRIPTION
**Pull request by bnmnetp**
Create a simple fixture that allows the tester to easily create an assignment, add questions, autograde, calculate totals set deadlines, etc.

Builds on work in #1257

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -542,3 +542,100 @@ def test_user(test_client, runestone_db_tools):
...
+        _TestAssignment.assignment_count += 1
+
+
+    def addq_to_assignment(self, **kwargs):
+        if 'points' not in kwargs:
+            kwargs['points'] = 1
+        kwargs['assignment'] = self.assignment_id
+        assert self.test_client.validate(
+            'admin/add__or_update_assignment_question', data=kwargs
+        ) != json.dumps('Error')
```
Comment (bjones1) [inline tests/conftest.py]: Seems fragile -- is there any way to measure this as an equality (verify that success or something similar was returned)? Here, even a return of `error` would still give a test passed.

File: tests/conftest.py [hunk 2]
```diff
@@ -542,3 +542,100 @@ def test_user(test_client, runestone_db_tools):
...
+        self.due = newdeadline
+        self.save_assignment()
+
+    def save_assignment(self):
+        assert json.loads(
+            self.test_client.validate('admin/save_assignment',
+            data=dict(assignment_id=self.assignment_id,
+                      visible='T' if self.is_visible else 'F',
+                      description=self.description,
+                      due=str(self.due)))) != "ERROR"
```
Comment (bjones1) [inline tests/conftest.py]: Again, avoid != for tests if possible.

File: tests/conftest.py (hunk 1 above)
Comment (bnmnetp) [reply tests/conftest.py]: I copied that straight out of your code.  :smile:

File: tests/conftest.py (hunk 2 above)
Comment (bjones1) [reply tests/conftest.py]: Oh no! Do as I say, not as I do... :)

File: tests/conftest.py (hunk 1 above)
Comment (bnmnetp) [reply tests/conftest.py]: The success indicator is that a dictionary of values is returned.  I could modify the dictionary to have a status key.  I was trying to keep this code simple and let the actual test take care of details.  For example the keys returned are total, activity_count, and activities_required.  That doesn't seem like something this method ought to worry about.

Comment (bnmnetp) [reply tests/conftest.py]: OK, I added a simple status key in both cases to indicate success.  No more `!=`

Comment (bjones1) [reply tests/conftest.py]: Cool!

**Pull request merged**